modules/sugar/src/main/ts/ephox/sugar/api/node/ShadowDom.ts

```diff
@@ -70,3 +70,31 @@ export const getShadowRoot = (e: Element<DomNode>): Option<Element<ShadowRoot>>
  */
 export const getShadowHost = (e: Element<ShadowRoot>): Element<DomElement> =>
   Element.fromDom(e.dom().host);
+
+/**
+ * When Events bubble up through a ShadowRoot, the browser changes the target to be the shadow host.
+ * This function gets the "original" event target if possible.
+ * This only works if the shadow tree is open - if the shadow tree is closed, event.target is returned.
+ * See: https://developers.google.com/web/fundamentals/web-components/shadowdom#events
+ */
+export const getOriginalEventTarget = (event: Event): Option<EventTarget> => {
+  if (isSupported()) {
```

Comment:
I wonder if to be on the safe side, we should only do this if the `event.target` is a shadow dom element? eg `isSupported() && isShadowRoot(event.target)`

Reply:
On the affected events, the event.target is the shadow _host_. I'll see.

Reply:
Ahh sorry I couldn't recall from the spec whether it was the root or host.

Reply:
Done. Lemme know what you think.